### PR TITLE
Added scala 3 support for akka based backend

### LINKED ISSUE #1225: Try enabling akka backend for Scala 3
https://akka.io/blog/news/2021/12/20/akka-2.6.18-released

### PR DESCRIPTION
**Pull request by Pask423**
closes #1225

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -446,7 +446,7 @@ lazy val akkaHttpBackend = (projectMatrix in file("akka-http-backend"))
   .settings(
     name := "akka-http-backend",
     libraryDependencies ++= Seq(
-      akkaHttp,
+      akkaHttp.cross(CrossVersion.for3Use2_13),
```
Comment (adamw) [inline build.sbt:449]: there's no akka http for scala 3?

Comment (Pask423) [reply build.sbt:449]: unfortunately no as now https://github.com/akka/akka-http/pull/3901

Comment (adamw) [reply build.sbt:449]: ah. Well if it works, that's the best we have. I've created https://github.com/softwaremill/sttp/issues/1241 to remove this in the future

Comment (adamw) [reply build.sbt:449]: ah2. I see the cross-version doesn't seem to work, so we'll have to keep this PR open until there's a new release.

Comment (liosedhel) [reply build.sbt:449]: Hi @adamw :)out of curiosity, what exactly is the problem here with akka-http for3Use2_13 workaround?

Comment (Pask423) [reply build.sbt:449]: There are some compilation errors with such setting, you can check CI run for more details
image

Comment (liosedhel) [reply build.sbt:449]: I see, it seems to be a problem of Akka dependency which is on provided scope in akka-http. akka-http expect _2.13 akka artefacts. You would have to probably add akka-stream also as for3Use2_13 🤔

**Comment by adamw** [comment]
akka-http won't be published for Scala 3 under an open-source license, so this unfortunately won't happen - closing.

**Pull request closed without merging**